bridges/snowbridge/primitives/core/src/lib.rs

```diff
@@ -163,4 +163,30 @@ impl DescribeLocation for DescribeHere {
 
 /// Creates an AgentId from a Location. An AgentId is a unique mapping to a Agent contract on
 /// Ethereum which acts as the sovereign account for the Location.
-pub type AgentIdOf = HashedDescription<H256, (DescribeHere, DescribeFamily<DescribeAllTerminal>)>;
+pub type AgentIdOf =
+	HashedDescription<AgentId, (DescribeHere, DescribeFamily<DescribeAllTerminal>)>;
+
+#[derive(Clone, Default, Eq, Debug, PartialEq, Ord, PartialOrd, Encode, Decode, TypeInfo)]
+pub struct AssetRegistrarMetadata {
+	pub name: Vec<u8>,
+	pub symbol: Vec<u8>,
+	pub decimals: u8,
+}
+
+pub type TokenId = H256;
+
+pub type TokenIdOf = HashedDescription<TokenId, DescribeGlobal>;
+
+pub struct DescribeGlobal;
+impl DescribeLocation for DescribeGlobal {
+	fn describe_location(l: &Location) -> Option<Vec<u8>> {
+		match l.unpack() {
```

Comment:
These cases seem incomplete. For example in the `ForeignAssets` pallet on AH there are foreign assets with locations of the form  `(Parent, Parachain(X)`. with no `GlobalConsensus`.

We should also have tests that exercise the various kinds of locations supported in  `ForeignAssets`

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/128/commits/009402e65b11c37956a0dc35024fbce3ddfafd01, 
the `(1,[GlobalConsensus(network)])` prefix is added by the reanchor logic in `limited_reserve_transfer_assets`, more context in
https://github.com/Snowfork/polkadot-sdk/pull/128#discussion_r1553737741

Reply:
Ah do you mean that when BH receives a PNA transfer instruction XCM from AH, the asset locations are modified to include a GlobalConsensus field?

Reply:
Exactly.